### ISSUE TITLE
Gutenberg doesn't prevent default html events

### ISSUE THREAD

**Issue by cawa-93** [WordPress/gutenberg#49355]
### Description

I tried to build simple block:

```html
<details>
	<summary><RitchText></summary>
</details>
```
The problem is that it is not possible to type a space character into the RitchText, because pressing the space-key trigger details to toggle, rather than typing a space character.

### Step-by-step reproduction instructions

1. Create block with markup:
```js
el('details', [
    el('summary', {}, [
        el(RichText, {
            tagName: 'span',
            placeholder: 'Try to type Space',

        })
    ]),
    'Content in details'
])
```
2. Add block to content
3. Try to write some text with spaces.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by carolinan** [comment]
Hi,
The keyboard events are disabled per block when needed.
Here is an example of how it is added to the pull request for the details block:
https://github.com/WordPress/gutenberg/blob/731a8e802bc2b0c7ba64c8fc1ef0bdaf91a0eee9/packages/block-library/src/details-summary/edit.js#L12

**Comment by cawa-93** [comment]
> Hi, The keyboard events are disabled per block when needed. Here is an example of how it is added to the pull request for the details block:
> 
> https://github.com/WordPress/gutenberg/blob/731a8e802bc2b0c7ba64c8fc1ef0bdaf91a0eee9/packages/block-library/src/details-summary/edit.js#L12

Sorry, but I don't really understand what you mean.

I tried add ` onClick={ ( event ) => event.preventDefault() } ` to `summary` in my block. In this case, pressing the spacebar does prevent toggling details, but at the same time the space character is not entered into `RichText`. I also tried prevent `onKeyUp` and got same effect - details doesn't toggle when I press spacebar, but space chapter doesn't entered. I tried prevent `onKeyDown` as well. In this case, no text is entering at all

**Comment by carolinan** [comment]
If you look closer at the linked pull request:
The `<summary>` is a wrapper for the richtext. The preventDefault is on the summary, not the richtext, this allows typing space in the richtext.
In the editor, the `<details>` is opened when selected:
https://github.com/WordPress/gutenberg/blob/731a8e802bc2b0c7ba64c8fc1ef0bdaf91a0eee9/packages/block-library/src/details/edit.js#L53


But in general I think it needs to be added per block, that Gutenberg should not prevent these events by default.
There are some more examples. Several links are prevented from being opened inside the editor:
https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/avatar/edit.js#L155

**Comment by cawa-93** [comment]
> The `<summary>` is a wrapper for the richtext. The preventDefault is on the summary, not the richtext, this allows typing space in the richtext.

Yeah. It's exacly what I did. My edit function looks like this:
```js
function edit({attributes, setAttributes}) {
    return el('details', {...useBlockProps()}, [
        el('summary', {
            onClick: (event) => event.preventDefault()
        }, [
            el(RichText, {
                value: attributes.label,
                onChange: (newSummary) =>
                    setAttributes({label: newSummary}),
                withoutInteractiveFormatting: true,

                placeholder: 'Try to type Space',
            })
        ]),
        'INNER CONTENT HERE'
    ])
}
```

**UPD** I just realized that the problem is not reproduced in Edge. Indeed, In Edge, you CAN type space. But, I test it in Firefox (112.0b7), and there, pressing spacebar doesn't type space chapter. [Bug tracker](https://bugzilla.mozilla.org/show_bug.cgi?id=1822860)

**Comment by ellatrix** [comment]
This was fixed in #30244 but removed later in #41977. Also, we now have a core details block and the solution there is to use a div inside the summary element. #45055

Ideally, browsers should disable interactivity when an element is content editable (like they do for a link), so I'd recommend reporting it to them.
